onmt/decoders/transformer.py

```diff
@@ -43,10 +44,34 @@ def __init__(self, d_model, heads, d_ff, dropout, attention_dropout,
         self.layer_norm_1 = nn.LayerNorm(d_model, eps=1e-6)
         self.layer_norm_2 = nn.LayerNorm(d_model, eps=1e-6)
         self.drop = nn.Dropout(dropout)
+        self.full_context_alignment = full_context_alignment
+        self.alignment_heads = alignment_heads
 
-    def forward(self, inputs, memory_bank, src_pad_mask, tgt_pad_mask,
-                layer_cache=None, step=None):
+    def forward(self, *args, **kwargs):
+        """ Extend _forward for multiply decoder pass.
```

Comment:
multiply --> multiple?
Maybe add one sentence to explain why we need it?

Reply:
How about this:
""" Extend _forward for (possibly) multiple decoder pass:
        1. Always a default (future masked) decoder forward pass,
        2. Possibly a second future aware decoder pass for joint learn
            full context alignement.
"""